### PR TITLE
LPS-151734

### PR DESCRIPTION
**Pull request by p-albuquerque**
Related Issue: https://issues.liferay.com/browse/LPS-151734

### CODE REVIEW

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @p-albuquerque, can you please check if these failures are related to your changes? Thanks

**Comment by liferay-forms** [comment]
Hey @p-albuquerque, can you please check if these failures are related to your changes? Thanks

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -452,6 +468,35 @@ public int getOneToManyRelatedObjectEntriesCount(
 		return objectEntryPersistence.dslQueryCount(dslQuery);
 	}
 
+	@Override
+	public List<Map<String, Object>> getOneToManySystemRelatedObjectEntries(
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java:472]: Could we rename to getOneToManyRelatedSystemObjectEntries ? I think that the "System" should be near to "ObjectEntries" ?

File: modules/apps/object/object-service/src/main/java/com/liferay/object/internal/related/models/ObjectEntry1to1ObjectRelatedModelsProviderImpl.java [hunk 2]
```diff
@@ -166,6 +167,16 @@ public int getRelatedModelsCount(
 		return relatedModels.size();
 	}
 
+	@Override
+	public List<Map<String, Object>> getRelatedSystemModels(
+			long foreignKey, ObjectDefinition objectDefinition,
+			ObjectRelationship objectRelationship)
+		throws PortalException {
+
+		return _objectEntryLocalService.getOneToManySystemRelatedObjectEntries(
```
Comment (marcelabc) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/internal/related/models/ObjectEntry1to1ObjectRelatedModelsProviderImpl.java:176]: could you explain me latter why are you using a method named 1toMany in class named for 1to1? Thanks :smile:

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi @p-albuquerque, great job!!! :rocket: 
We can discuss together the questions and the changes requested!

File: modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/entries/frontend/data/set/data/model/RelatedModel.java [hunk 3]
```diff
@@ -32,7 +40,13 @@ public String getLabel() {
 		return _label;
 	}
 
+	public Boolean getSystem() {
```
[the remark this answers is not in the document]
Comment (p-albuquerque) [reply modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/entries/frontend/data/set/data/model/RelatedModel.java:43]: I will fix this. Thanks!

**Comment by p-albuquerque** [comment]
Note:
- Check sort parameters

**Comment by p-albuquerque** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by liferay-forms** [comment]
Hey @p-albuquerque, can you please check if these failures are related to your changes? Thanks

**Comment by marcelabc** [comment]
Please see https://github.com/liferay-objects/liferay-portal/pull/603
@p-albuquerque

**Pull request closed without merging**